uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/FixNotMavenizedArtifactInfo.java

```diff
@@ -0,0 +1,61 @@
+package org.guvnor.m2repo.backend.server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.guvnor.common.services.project.model.GAV;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.ARTIFACT_ID;
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.GROUP_ID;
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.VERSION;
+
+public class FixNotMavenizedArtifactInfo {
+
+    private static final Logger logger = LoggerFactory.getLogger(FixNotMavenizedArtifactInfo.class);
+
+    private final Properties notMavenizedArtifacts = new Properties();
+    private final String pomTemplate;
+
+    public FixNotMavenizedArtifactInfo() {
+        String pomTemplate = null;
+        try {
+            pomTemplate = new BufferedReader(new InputStreamReader(M2ServletContextListener.class.getResourceAsStream("template.pom")))
+                    .lines().collect(Collectors.joining("\n"));
+            final InputStream isNotMavenizedArtifacts = M2ServletContextListener.class.getResourceAsStream("/not-mavenized-artifacts.txt");
+            if (isNotMavenizedArtifacts != null) {
+                notMavenizedArtifacts.load(isNotMavenizedArtifacts);
+            }
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+        }
+        this.pomTemplate = pomTemplate;
+    }
+
+    public Properties getProperties(final String filePath) {
+        final Properties result = new Properties();
+        final String fullFileName = filePath.substring(filePath.lastIndexOf("/") + 1);
+        final String fileNamePart = fullFileName.substring(0, fullFileName.indexOf("-"));
+        final String fileVersion = fullFileName.substring(fullFileName.indexOf("-") + 1).replace(".jar", "");
```

Comment:
This code for extracting file version looks very brittle. Even in this PR ensemble I see jar filenames like
`uberfire-m2repo-editor-backend-100-SNAPSHOT.jar` not to mention that you can have "1.3.2-SNAPSHOT" that also contains dash. You also have for example `ant-launcher` in one of the not-mavenized-artifacts.txt for which `fullFileName.substring(0, fullFileName.indexOf("-"));` will return `ant` -> will not find it in the properties for this artifacts. This should be made more robust and unit tested thoroughly. Please add cases like that to unit tests (fileNamePart containing multiple dashes, fileVersion containing SNAPSHOT etc).